File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays.Python/director/scores/data/ray_score_interval_descriptor.py
class RayScoreIntervalDescriptor:
    """Frame descriptor for sprite timeline intervals."""
    def __init__(self):
        self.start_frame = 0
        self.end_frame = 0
        self.unknown1 = 0
        self.unknown2 = 0
        self.sprite_number = 0
        self.unknown_always_one = 0
        self.unkown_a = 0
        self.unkown_b = 0
        self.unknown_e1 = 0
        self.unknown_fd = 0
        self.unknown7 = 0
        self.unknown8 = 0
        self.extra_values = []
        self.channel = 0
        self.behaviors = []
        self.flip_h = False
        self.flip_v = False
        self.editable = False
        self.moveable = False
        self.trails = False
        self.is_locked = False

File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays.DotNet/director/Scores/Data/RayScoreIntervalDescriptor.cs
using static ProjectorRays.director.Scores.RaysScoreChunk;

namespace ProjectorRays.director.Scores.Data;

internal class RayScoreIntervalDescriptor
{
    public int StartFrame { get; internal set; }
    public int EndFrame { get; internal set; }
    public int Unknown1 { get; internal set; }
    public int Unknown2 { get; internal set; }
    public int SpriteNumber { get; internal set; }
    public int UnknownAlwaysOne { get; internal set; }
    public int UnkownA { get; internal set; }
    public int UnkownB { get; internal set; }
    public int UnknownE1 { get; internal set; }
    public int UnknownFD { get; internal set; }
    public int Unknown7 { get; internal set; }
    public int Unknown8 { get; internal set; }
    public List<int> ExtraValues { get; } = new();
    public int Channel { get; internal set; }
    public List<RaysBehaviourRef> Behaviors { get; internal set; } = new List<RaysBehaviourRef>();
    public bool FlipH { get; internal set; }
    public bool FlipV { get; internal set; }
    public bool Editable { get; internal set; }
    public bool Moveable { get; internal set; }
    public bool Trails { get; internal set; }
    public bool IsLocked { get; internal set; }
}

